commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitXMLReporter.java

```diff
@@ -174,121 +156,79 @@ protected void generateReport(ISuite suite) {
             //            }
          }
       }
-      Properties attrs = new Properties();
-      attrs.setProperty(XMLConstants.ATTR_TESTS, "" + testCount);
-      attrs.setProperty(XMLConstants.ATTR_TIME, "" + elapsedTime / 1000.0);
-      attrs.setProperty(XMLConstants.ATTR_NAME, getModuleSuffix());
-      attrs.setProperty("skipped", "" + m_numSkipped);
-      attrs.setProperty(XMLConstants.ATTR_ERRORS, "0");
-      attrs.setProperty(XMLConstants.ATTR_FAILURES, "" + (m_numFailed.get()));
-      document.push(XMLConstants.TESTSUITE, attrs);
-      showProperties(document);
-
-      document.addComment("Tests results");
-      createElementFromTestResults(document, m_allTests.values());
 
-      document.pop();
-
-      // Reset output directory
-      Utils.writeUtf8File(suite.getOutputDirectory().replaceAll(".Surefire suite", ""),
-            generateFileName(suite) + ".xml", document.toXML());
+      PolarionJUnitXMLWriter writer = null;
+      try {
+         String outputDir = suite.getOutputDirectory().replaceAll(".Surefire suite", "");
+         String baseName = String.format("TEST-%s.xml", getSuiteName(suite));
+         File outputFile = new File(outputDir, baseName);
+         writer = new PolarionJUnitXMLWriter(outputFile);
+         writer.start(getModuleSuffix(), testCount, m_numSkipped.get(), m_numFailed.get(), elapsedTime, true);
+
+         // TODO Also write a report based on suite.getAllInvokedMethod() and check for differences
```

Comment:
Is this meant to be addressed in this PR?

Reply:
Nope, it just isn't clear why we need to keep track of the executed tests ourselves instead of using TestNG's built-in results